rectools/visuals/metrics_app.py

```diff
@@ -0,0 +1,128 @@
+
+    def _validate_input_data(self) -> None:
+        if not isinstance(self.metrics_data, pd.DataFrame):
+            raise ValueError("Incorrect input type. `metrics_data` should be a DataFrame")
+        if Columns.Split not in self.metrics_data.columns:
+            raise KeyError("Missing 'Split' column in `metrics_data` DataFrame")
+        if Columns.Model not in self.metrics_data.columns:
+            raise KeyError("Missing 'Model' column in `metrics_data` DataFrame")
+        if len(self.metrics_data.columns) < 3:
+            raise KeyError("`metrics_data` DataFrame assumed to have at least one metric column")
+
+    def _validate_input_patameters(self) -> None:
+        if self.chart_width < 10:
+            raise ValueError(
+                "Incorrect `chart_width` value. `chart_width` should be a float or int in the interval [10, inf]"
+            )
+        if self.chart_height < 10:
+            raise ValueError(
+                "Incorrect `chart_height` value. `chart_height` should be a float or int in the interval [10, inf]"
+            )
+
+    def _get_metric_names(self) -> tp.List[str]:
+        non_metric_columns = {Columns.Split, Columns.Model}
+        return [col for col in self.metrics_data.columns if col not in non_metric_columns]
+
+    def _get_folds_number(self) -> int:
+        return self.metrics_data[Columns.Split].nunique()
+
+    def _make_chart_data(self, fold_number: int) -> pd.DataFrame:
+        return self.metrics_data[self.metrics_data[Columns.Split] == fold_number].drop(columns=Columns.Split)
+
+    def _make_chart_data_avg(self) -> pd.DataFrame:
+        return self.metrics_data.drop(columns=Columns.Split).groupby(Columns.Model, sort=False).mean().reset_index()
+
+    def _create_chart(self, avg_folds: bool, fold_number: int, metric_x: str, metric_y: str) -> go.Figure:
+        chart_data = self._make_chart_data_avg() if avg_folds else self._make_chart_data(fold_number)
+        fig = px.scatter(
+            chart_data,
+            x=metric_x,
+            y=metric_y,
+            color=Columns.Model,
+            width=self.chart_width,
+            height=self.chart_height,
+            color_discrete_sequence=self.color_discrete_sequence,
+        )
+        fig.layout.update(showlegend=self.show_legend)
+        return fig
+
+    def display(self) -> None:
+        """TODO: add dosctring"""
+        metrics_list = self._get_metric_names()
+        n_splits = self._get_folds_number()
+
+        metric_x = widgets.Dropdown(description="Metric X:", value=metrics_list[0], options=metrics_list)
+        metric_y = widgets.Dropdown(description="Metric Y:", value=metrics_list[-1], options=metrics_list)
+        container_metrics = widgets.HBox(children=[metric_x, metric_y])
+
+        use_avg = widgets.Checkbox(description="Avg folds", value=True)
+        fold_number = widgets.Dropdown(description="Fold number:", value=0, options=list(range(n_splits)))
```

Comment:
It's better to hide this field somehow if `Avg folds` is selected.

Or we can make a single Dropdown `Fold number` with options `Avg, 0, 1, 2, ...`

Reply:
Done